fe/fe-core/src/main/java/com/starrocks/sql/parser/SqlParser.java

```diff
@@ -88,7 +90,28 @@ public static StatementBase parseWithOldParser(String originStmt, long sqlMode,
         }
     }
 
+
+    /*
+     * The new version of parser can handle comments (discarded directly in lexical analysis).
+     * But there are some special cases when the old and new parsers are compatible.
+     * Because the parser does not support all statements, we split the sql according to ";".
+     * For example, sql: --xxx;\nselect 1; According to the old version, it will be parsed into one statement,
+     * but after splitting according to the statement,
+     * two statements will appear. This leads to incompatibility.
+     * Because originSql is stored in the old version of the materialized view, it is the index stored in the old version.
+     * */
+    static String regexSimpleComment = "--.*\n";
+    static Pattern psm = Pattern.compile(regexSimpleComment);
+
+    static String regexBracketComment = "/\\* .*? \\*/";
+    static Pattern pbc = Pattern.compile(regexBracketComment);
+
     private static List<String> splitSQL(String sql) {
+        Matcher m = psm.matcher(sql);
```

Comment:
Would it make the comment disappear in audit log and other logs?

Reply:
I don't think so. The original SQL is stored in logs as well as XXStmt class.

Reply:
no, only for parse